nix/nixos/typhon.nix

```diff
@@ -31,13 +31,29 @@ in {
       default = "/var/lib/typhon";
       description = "Home directory for the Typhon instance";
     };
-    passwordFile = mkOption {
-      type = types.str;
-      description = "Path to a file containing the admin password";
+    hashedPassword = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "The hash of the admin password";
+    };
+    hashedPasswordFile = mkOption {
+      type = types.nullOr types.str;
+      default =
+        if cfg.hashedPassword == null
+        then null
+        else builtins.toString (pkgs.writeText "typhon-password" cfg.hashedPassword);
+      description = "Path to a file containing the hash of the admin password";
```

Comment:
i guess here we should say the "argon2 hash of the admin password", not just "the hash", right?